Tests/NIOHTTP2Tests/ConfiguringPipelineAsyncMultiplexerTests.swift

```diff
@@ -0,0 +1,150 @@
+            context.channel.writeAndFlush(requestFramePayload, promise: nil)
+        }
+
+        func channelActive(context: ChannelHandlerContext) {
+            self.writeRequest(context: context)
+            context.fireChannelActive()
+        }
+
+        func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+            let frame = self.unwrapInboundIn(data)
+            switch frame {
+            case .headers:
+                break
+            default:
+                fatalError("unexpected frame type: \(frame)")
+            }
+            context.fireChannelRead(data)
+        }
+    }
+
+    func testBasicPipelineCommunicates() async throws {
+        let requestCount = 100
+
+        let serverRecorder = InboundFramePayloadRecorder()
+        let (_, clientMultiplexer) = try await assertNoThrowWithValue(
+            try await self.clientChannel.configureHTTP2PipelineAsync(
+                mode: .client, connectionConfiguration: .init(), streamConfiguration: .init()) { channel in self.serverChannel.eventLoop.makeSucceededFuture(channel) }.get()
+        )
+
+        let (serverStreamChannels, _) = try await assertNoThrowWithValue(
+            try await self.serverChannel.configureHTTP2PipelineAsync(
+                mode: .server, connectionConfiguration: .init(), streamConfiguration: .init()) { channel in
+                    channel.pipeline.addHandlers([OKResponder(), serverRecorder]).map { _ in channel }
+                }.get()
+        )
+
+        try await assertNoThrow(try await self.assertDoHandshake(client: self.clientChannel, server: self.serverChannel))
+
+
+        try await withThrowingTaskGroup(of: Int.self, returning: Void.self) { group in
+            // server
+            group.addTask {
+                var serverInboundChannelCount = 0
+                for try await _ in serverStreamChannels {
+                    serverInboundChannelCount += 1
+                }
+                return serverInboundChannelCount
+            }
+
+            // client
+            for _ in 0 ..< requestCount {
+                // Let's try sending some requests
+                let streamChannel = try await clientMultiplexer.createStreamChannel() { channel in
+                    return channel.pipeline.addHandlers([SimpleRequest(), InboundFramePayloadRecorder()]).map {
+                        return channel
+                    }
+                }
+
+                let clientRecorder = try await streamChannel.pipeline.handler(type: InboundFramePayloadRecorder.self).get()
+                try await self.interactInMemory(self.clientChannel, self.serverChannel)
+
+                await self.clientChannel.testingEventLoop.run()
+                try await streamChannel.closeFuture.get()
+
+                clientRecorder.receivedFrames.assertFramePayloadsMatch([ConfiguringPipelineAsyncMultiplexerTests.responseFramePayload])
+            }
+
+            async let closeFuture: Void = self.clientChannel.close()
```

Comment:
Do you have a particular reason for using an `async let` here? If the reason is because you need to run the EL to make the close actually close then just use the `close(promise: nil)` variant instead.

Reply:
It _used_ to be required for the test to pass but changing the recorders to use streams seems to have fixed that issue